src/ConnectionAutomaton.cpp

```diff
@@ -60,8 +60,7 @@ void ConnectionAutomaton::connect(
     std::shared_ptr<FrameTransport> frameTransport,
     bool sendingPendingFrames) {
   CHECK(isDisconnectedOrClosed());
-  CHECK(frameTransport);
-  CHECK(!frameTransport->isClosed());
+  CHECK(frameTransport && !frameTransport->isClosed());
```

Comment:
the check violation should be caught during testing and development. I don't expect this to be ever hit in the production so my thinking was having one check statement generates less code and less overhead.